source/common/router/router.cc

```diff
@@ -1754,6 +1755,10 @@ void Filter::onUpstreamMetadata(Http::MetadataMapPtr&& metadata_map) {
 
 void Filter::onUpstreamComplete(UpstreamRequest& upstream_request) {
   if (!downstream_end_stream_) {
+    if (allow_multiplexed_upstream_half_close_) {
+      // Continue request if downstream is not done yet.
+      return;
```

Comment:
While we shouldn't be resetting the upstream, we do also skip over doing these various timing, outlier detection, etc. When will we end up updating them in this case when `allow_multiplied_upstream_half_close` is true?

Reply:
Never. This code updates the `response_time` which is `response_complete_time - request_complete_time `. It is negative for the case where response completes before request and budget percentage computations will be off. Histograms record uint64_t so these will end up being some large values.

Since the `response_time` becomes nonsensical in this case, I decided to skip recording the stats.